webpack.config.js

```diff
@@ -0,0 +1,41 @@
+const _ = require('lodash');
+const path = require('path');
+const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+const { tokens } = require('./lib/dsl/lexer');
+
+const allTokenTypeNames = _.mapValues(tokens, tokenType => tokenType.name);
+
+module.exports = {
+  mode: 'production',
+  entry: './lib/dsl/exports.js',
+  output: {
+    path: path.resolve(__dirname, './dist/'),
+    filename: 'jhipster.min.js',
+    library: 'jdl',
+    libraryTarget: 'umd',
+    // https://github.com/webpack/webpack/issues/6784
+    globalObject: 'typeof self !== \'undefined\' ? self : this'
+  },
+  externals: {
+    // https://webpack.js.org/guides/author-libraries/#externalize-lodash
+    lodash: {
+      commonjs: 'lodash',
```

Comment:
is lodash the only dependency that we are bundling?

Reply:
lodash is not bundled here, it is defined as external which means it is resolved at runtime.

Reply:
Sorry, yes I know, wrong choice of word. what I meant was to ask if lodash was the only dependency and I checked the chevrotain package.json after writing this. Thanks anyway

Reply:
Chevrotain has a dependency to regexp-to-ast npm package (a regexp parser I wrote).
But I make sure to bundle it inside the chevrotain.min.js bundle for ease of consumption in the browser so you don't have to load anything extra except lodash.

You can see this in the sample html mocha runner test provided in this commit

Reply:
Actually it does not matter because Chevrotain is bundled inside the JDL-core.js bundle
so webpack handles the transitive dependencies here.